### PR TITLE
[CategoryAxis] Make rotated ticks wrap less aggressively

### LINKED ISSUE #1526: Rotated ticks wrap aggressively
screen shot 2015-01-16 at 6 18 45 pm
Repro: http://jsfiddle.net/xcgjf426/1/

### PR DESCRIPTION
**Pull request by rcchen**
It appears that the crux of the matter is that SVGTypewriter does not really work for rotated text as far as I know, so in cases where the text is rotated, I am taking an approach where adjustments to width requested are made based on the angle of the tick label, and then flipping the width and height results at the end of _measureTicks.

JSFiddle of fix: http://jsfiddle.net/xcgjf426/3/

Fix #1526

### CODE REVIEW

File: src/components/axes/categoryAxis.ts [hunk 1]
```diff
@@ -138,22 +138,51 @@ export module Axis {
...
       };
+
+      // return {
+      //   textFits: wrappingResults.every((t: SVGTypewriter.Wrappers.WrappingResult) =>
+      //               SVGTypewriter.Utils.StringMethods.isNotEmptyString(t.truncatedText) && t.noLines === 1),
+      //   usedWidth : widthFn<SVGTypewriter.Wrappers.WrappingResult, number>(wrappingResults,
+      //                 (t: SVGTypewriter.Wrappers.WrappingResult) => this._measurer.measure(t.wrappedText).width, 0),
+      //   usedHeight: heightFn<SVGTypewriter.Wrappers.WrappingResult, number>(wrappingResults,
+      //                 (t: SVGTypewriter.Wrappers.WrappingResult) => this._measurer.measure(t.wrappedText).height, 0)
+      // };
```
Comment (jtlan) [inline src/components/axes/categoryAxis.ts]: Remove this commented-out code

**Comment by jtlan** [comment]
Would it be easier to fix this in [svg-typewriter](https://github.com/palantir/svg-typewriter)?

**Comment by rcchen** [comment]
Discussed offline with @jtlan about shipping this feature now, then going back to svg-typewriter to address the actual issue. Tracking at https://github.com/palantir/svg-typewriter/issues/25

File: src/components/axes/categoryAxis.ts [hunk 2]
```diff
@@ -138,21 +138,42 @@ export module Axis {
...
       var heightFn = this._isHorizontal() ? _Util.Methods.max : d3.sum;
 
+      var textFits = wrappingResults.every((t: SVGTypewriter.Wrappers.WrappingResult) =>
+                    SVGTypewriter.Utils.StringMethods.isNotEmptyString(t.truncatedText) && t.noLines === 1);
+      var usedWidth = widthFn<SVGTypewriter.Wrappers.WrappingResult, number>(wrappingResults,
+                      (t: SVGTypewriter.Wrappers.WrappingResult) => this._measurer.measure(t.wrappedText).width, 0);
+      var usedHeight = heightFn<SVGTypewriter.Wrappers.WrappingResult, number>(wrappingResults,
+                      (t: SVGTypewriter.Wrappers.WrappingResult) => this._measurer.measure(t.wrappedText).height, 0);
+
+      // If the tick labels are rotated, reverse usedWidth and usedHeight
```
Comment (jtlan) [inline src/components/axes/categoryAxis.ts]: maybe apply the HACKHACK here too?

**Comment by jtlan** [comment]
Unit test?

**Comment by crmorford** [comment]
These really long words should be wrapping
http://jsfiddle.net/xcgjf426/4/

**Comment by rcchen** [comment]
Updated JSFiddle: http://jsfiddle.net/xcgjf426/5/

**Pull request merged**